contracts/contracts/multisig/util.cairo

```diff
@@ -1,6 +1,46 @@
-func almost_equal(a, b) -> (res: felt):
+from starkware.cairo.common.math import assert_not_zero
+from starkware.cairo.common.default_dict import default_dict_new, default_dict_finalize
+from starkware.cairo.common.dict import dict_update
+from starkware.cairo.common.dict_access import DictAccess
+
+func almost_equal(a, b) -> (res : felt):
     if (a - b) * (a - b - 1) * (a - b + 1) == 0:
         return (res=1)
     end
     return (res=0)
 end
+
+func assert_unique_addresses_impl{range_check_ptr, dict_ptr : DictAccess*}(
+    index : felt, data_len : felt, data : felt*
+):
+    alloc_locals
+    if index == data_len:
+        return ()
+    end
+
+    local el = data[0]
+    assert_not_zero(el)
```

Comment:
This will fail, and then the error message "owners not unique" will be returned.

I suggest we remove this altogether and allow 0x0 addresses so we could burn multisig ownership by `set_owners_and_threshold(owners=[0x0], threshold=1).

Looks like Serum doesn't check for 0x0 owners. cc @archseer

Reply:
Just to give an idea behind what is written in that function:
We have dictionary where all `dict[key]` are set to `default_value`. 
`default_dict_new()`: `Returns a new dictionary where all keys are initialized with a given default value.`
Then for each key we update it's previous value(`default_value`) to `owners[i]`. If in `dict_update` incorrect `prev_value`  was passed then cairo will panic. So if `dict[key]` was updated at least once we will panic thus detecting repeating value.

Check on L22 is necessary because otherwise such array will be possible `[default_value, default_value, ....]`
`dict_new` wasn't used since its not present on StarkNet bacause it requires hints for initialization.

So we can whether create new magic `default_value` like -1 or use different approach for detecting repeating address(`usort` builtin sort doesn't work).

PS. to future myself not to forget substitute
`assert_not_zero(el - default_value)`

Reply:
In the aggregator I've used `key=address` `value=owners_index+1`, using 1-indexed values for `_owners` instead of 0-indexed. That way `0` means undefined and addresses are stored with indexes `1..N`

Reply:
Used `array.length` as `default_value` to avoid mess with `1..N`